### PR TITLE
Adding SParseIntegerSet method

### PR DESCRIPTION
**Pull request by danieldoglas**
### Details

This PR adds a new method to transform a list into a SET. 

### Tests

_________
**Internal Testing Reminder:** when changing bedrock, please compile auth against your new changes

### CODE REVIEW

**Review by tylerkaraszewski — APPROVED** [review]
It would be nice not to add thgese methods that do everything twice (i.e., create a list and then create a similar set, or create a list of strings and then create a similar list of integers), but I don't think the real-world performance penalty is enough to spend the time or code complexity on it.

File: libstuff/libstuff.cpp [hunk 1]
```diff
@@ -654,6 +654,16 @@ list<int64_t> SParseIntegerList(const string& value, char separator) {
     return valueList;
 }
 
+// --------------------------------------------------------------------------
+set<int64_t> SParseIntegerSet(const string& value, char separator) {
+    set<int64_t> valueSet;
+    list<string> strings = SParseList(value, separator);
+    for (string str : strings) {
```
Comment (tylerkaraszewski) [inline libstuff/libstuff.cpp:661]: Oops, at least use `const string & str` here, it will avoid making an extra copy of each string.

**Review by tylerkaraszewski — CHANGES_REQUESTED** [review]

File: libstuff/libstuff.cpp (hunk 1 above)
Comment (danieldoglas) [reply libstuff/libstuff.cpp]: Fixed!

**Review by tylerkaraszewski — APPROVED** [review]

**Comment by danieldoglas** [comment]
I'm gonna go ahead and merge this so I unblock my Auth PR

**Comment by aldo-expensify** [comment]
Nice! I have wanted to have `SParseIntegerSet` a few times :)

**Pull request merged**